### DISCUSSION TITLE
I tired to add response Interceptor, and then I got a type error.

### DISCUSSION THREAD

**Discussion by JohnYang02182** [axios/axios#5660] [Q&A]
I tried to add a response Interceptor like below
```js
axios.interceptors.response.use((res: AxiosResponse) => {
  if (res.status === 200) {
    if (res.data.code !== '993') {
      return Promise.resolve(res)
    } else {
      console.log('overtime')
    }
  } else {
    return Promise.reject(res)
  }
})
```

And then I got the error messages--
"The argument of type '(res: AxiosResponse) => Promise<AxiosResponse<any, any>> | undefined' cannot be assigned to the parameter of type '(value: AxiosResponse<any, any>) => AxiosResponse<any, any> | Promise<AxiosResponse<any, any>>'."

My version:
typescript : "4.9.3"
axios: "1.3.4"

I would appreciate it if someone could tell me what to do, or if something I missed.

**Comment by NotEvenANeko** [comment]
You need to return a `AxiosResponse<any, any> | Promise<AxiosResponse<any, any>>` here. Otherwise it will return nothing which is `undefined` when `res.status === 200` and `res.data.code === '993'`.

```typescript
axios.interceptors.response.use((res: AxiosResponse) => {
  if (res.status === 200) {
    if (res.data.code !== '993') {
      return Promise.resolve(res)
    } else {
      console.log('overtime') 
      // ^^^^^^^^^^^^^^
    }
  } else {
    return Promise.reject(res)
  }
})
```
